Unable to compile under Windows

I try to install under windows, but it won't compile.

src/python-lz4.c(12) : error C2146: syntax error : missing ')' before identifier
 'x'
src/python-lz4.c(12) : error C2081: 'uint32_t' : name in formal parameter list i
llegal
src/python-lz4.c(12) : error C2061: syntax error : identifier 'x'
src/python-lz4.c(12) : error C2059: syntax error : ';'
src/python-lz4.c(12) : error C2059: syntax error : ')'
src/python-lz4.c(13) : error C2449: found '{' at file scope (missing function he
ader?)
src/python-lz4.c(18) : error C2059: syntax error : '}'
src/python-lz4.c(25) : error C2065: 'uint32_t' : undeclared identifier

It seems like uint32_t not available under windows.

100% same problem to me :(

My output from easy_install, manually compiling give my same output

src/python-lz4.c(12) : error C2146: syntax error : missing ')' before identifier
 'x'
src/python-lz4.c(12) : error C2081: 'uint32_t' : name in formal parameter list i
llegal
src/python-lz4.c(12) : error C2061: syntax error : identifier 'x'
src/python-lz4.c(12) : error C2059: syntax error : ';'
src/python-lz4.c(12) : error C2059: syntax error : ')'
src/python-lz4.c(13) : error C2449: found '{' at file scope (missing function he
ader?)
src/python-lz4.c(18) : error C2059: syntax error : '}'
src/python-lz4.c(25) : error C2065: 'uint32_t' : undeclared identifier
src/python-lz4.c(44) : warning C4013: 'store_le32' undefined; assuming extern re
turning int
src/python-lz4.c(73) : error C2065: 'uint32_t' : undeclared identifier
src/python-lz4.c(73) : error C2146: syntax error : missing ';' before identifier
 'dest_size'
src/python-lz4.c(73) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(82) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(82) : warning C4013: 'load_le32' undefined; assuming extern ret
urning int
src/python-lz4.c(83) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(85) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(88) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(89) : error C2065: 'dest_size' : undeclared identifier
src/python-lz4.c(91) : error C2065: 'dest_size' : undeclared identifier
error: Setup script exited with error: command '"C:\Program Files\Microsoft Visu
al Studio 9.0\VC\BIN\cl.exe"' failed with exit status 2

--- a/src/python-lz4.c
+++ b/src/python-lz4.c
@@ -202,10 +202,11 @@ initlz4(void)
 #else
     PyObject *module = Py_InitModule("lz4", Lz4Methods);
 #endif
+    struct module_state *st = NULL;
 
     if (module == NULL)
         INITERROR;
-    struct module_state *st = GETSTATE(module);
+    st = GETSTATE(module);
 
     st->error = PyErr_NewException("lz4.Error", NULL, NULL);
     if (st->error == NULL) {

--- a/src/python-lz4.h
+++ b/src/python-lz4.h
@@ -42,6 +42,16 @@ PyMODINIT_FUNC initlz4(void);
 
 #if defined(_WIN32) && defined(_MSC_VER)
 # define inline __inline
+# if _MSC_VER >= 1600
+#  include <stdint.h>
+# else /* _MSC_VER >= 1600 */
+   typedef signed char       int8_t;
+   typedef signed short      int16_t;
+   typedef signed int        int32_t;
+   typedef unsigned char     uint8_t;
+   typedef unsigned short    uint16_t;
+   typedef unsigned int      uint32_t;
+# endif /* _MSC_VER >= 1600 */
 #endif
 
 #if defined(__SUNPRO_C) || defined(__hpux) || defined(_AIX)

